src/peft/utils/other.py

```diff
@@ -1556,3 +1554,45 @@ def create_attention_mask(
             position_ids=position_ids,
         )
     return attention_mask
+
+
+def _get_module_names_tied_with_embedding(model):
+    """
+    Get the list of the fully qualified names of the modules that are tied to the input embeddings. In case of a
+    source-target-mapping `_tied_weights_keys`, it will attempt to identify the input embedding weights.
```

Comment:
I find the second sentence confusing ("In case of"), it makes it sound like if it's a dict, the input embedding will be returned, but it means it's still the target that is returned, but the function tries to identify only those targets that are actually tied to the input embedding.

Also, let's mention that this function is for compatibility with <v5 and >= v5. Let's add a TODO for when <v5 is no longer supported.